app/notes/notes.service.js

```diff
@@ -1,69 +1,75 @@
 (function() {
-  angular.module('meganote.notes')
+  angular
+    .module('meganote.notes')
     .service('NotesService', NotesService);
 
   NotesService.$inject = ['$http'];
   function NotesService($http) {
-    var service = this;
-    service.notes = [];
+    var vm = this;
+    vm.notes = [];
+    vm.create = create;
+    vm.delete = delete(notes);
```

Comment:
This is not a reference to a function, but rather calls the delete function with the parameter notes and return the result of that function. This should probably be changed to something like `service.deleteNote = deleteNote`